src/Utils/featureFlags.tsx

```diff
@@ -0,0 +1,78 @@
+import { createContext, useContext, useState, useEffect } from "react";
+import useQuery from "./request/useQuery";
+import routes from "../Redux/api";
+import useAuthUser from "../Common/hooks/useAuthUser";
+import { FacilityModel } from "../Components/Facility/models";
+
+export type FeatureFlag = "SCRIBE_ENABLED"; // "HCX_ENABLED" | "ABDM_ENABLED" |
+
+export interface FeatureFlagsResponse {
+  user_flags: FeatureFlag[];
+  facility_flags: {
+    facility: string;
+    features: FeatureFlag[];
+  }[];
+}
+
+const defaultFlags: FeatureFlag[] = [];
+
+const FeatureFlagsContext = createContext<FeatureFlagsResponse>({
+  user_flags: defaultFlags,
+  facility_flags: [],
+});
+
+export const FeatureFlagsProvider = (props: { children: React.ReactNode }) => {
+  const [featureFlags, setFeatureFlags] = useState<FeatureFlagsResponse>({
+    user_flags: defaultFlags,
+    facility_flags: [],
+  });
+
+  const user = useAuthUser();
+
+  useEffect(() => {
+    if (user.user_flags) {
+      setFeatureFlags((ff) => ({
+        ...ff,
+        user_flags: [...defaultFlags, ...(user.user_flags || [])],
+      }));
+    }
+  }, [user]);
+
+  return (
+    <FeatureFlagsContext.Provider value={featureFlags}>
+      {props.children}
+    </FeatureFlagsContext.Provider>
+  );
+};
+
+export const useFeatureFlags = (facility?: FacilityModel | string) => {
```

Comment:
How about having it in a way that, it can be used in the following way instead:

This way from the place where this is used, which mostly always performs `.includes` check to check if a flag is present or not need not be repeated.

```js
const scribeEnabled = useFeatureFlag("SCRIBE_ENABLED", {...});
```

Reply:
@rithviknishad We can, but components where we required N flags would need N `useFeatureFlag`s, which will perform N requests for flags. It can of course be prevented if you specify the facility object instead of the ID to the hook, but it would just add overhead to the parent component in places where only the string is available. Might get in the way for DevX 😅